TheWarWithin/DeathKnightBlood.lua

```diff
@@ -1187,7 +1185,14 @@ spec:RegisterHook( "reset_precast", function ()
         state:QueueAuraExpiration( "vampiric_blood", TriggerUmbilicusEternus, buff.vampiric_blood.expires )
     end
 
-    if talent.vampiric_strike.enabled and IsActiveSpell( 433899 ) then applyBuff( "vampiric_strike" ) end
+    if talent.infliction_of_sorrow.enabled and buff.gift_of_the_sanlayn.up then
+        state:QueueAuraExpiration( "gift_of_the_sanlayn", TriggerInflictionOfSorrow, buff.gift_of_the_sanlayn.expires )
+    end
+
+    if IsActiveSpell( 433899 ) or IsActiveSpell( 433895 ) then
+        applyBuff( "vampiric_strike" )
+        class.abilities[ 433895 ] = class.abilities.heart_strike
```

Comment:
Why here?  This normally goes in the `copy` line of the ability model (`RegisterAbilities`).

Reply:
You're right, that should be enough. I was mostly just copy-pasting from the unholy implementation for speed.

Reply:
I'll just delete that assignment and verify everything still works.